src/components/SegmentedControl/component.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { useState, useCallback } from 'react';
+
+import { Testable, useBuildTestId } from '../../modules/test-ids';
+
+import { Container, Element, Size } from './styled';
+
+export type Props = Omit<React.AllHTMLAttributes<HTMLElement>, 'as' | 'size'> &
+  Testable & {
+    options: string[];
```

Comment:
```suggestion
    options: React.ReactNode[];
```

So that we can use `<FormattedMessage />` from `react-intl` directly.